remoting/subsystem/src/main/java/org/jboss/as/remoting/RemoteOutboundConnectionService.java

```diff
@@ -145,15 +145,12 @@ public String getProtocol() {
      * @throws URISyntaxException
      */
     private synchronized URI getConnectionURI() throws IOException, URISyntaxException {
-        if (this.connectionURI != null) {
-            return this.connectionURI;
-        }
+        /* WFCORE-851 - do not cache connectionURI else reconnect will fail if DNS changes */
         final OutboundSocketBinding destinationOutboundSocket = this.destinationOutboundSocketBindingInjectedValue.getValue();
         final InetAddress destinationAddress = destinationOutboundSocket.getResolvedDestinationAddress();
         final int port = destinationOutboundSocket.getDestinationPort();
 
-        this.connectionURI = new URI(protocol + "://" + NetworkUtils.formatPossibleIpv6Address(destinationAddress.getHostAddress()) + ":" + port);
-        return this.connectionURI;
+        return new URI(protocol + "://" + NetworkUtils.formatPossibleIpv6Address(destinationAddress.getHostAddress()) + ":" + port);
```

Comment:
The connectionURI field should also be removed altogether, as it is no longer used.